src/reason-parser/reason_pprint_ast.ml

```diff
@@ -2334,17 +2334,31 @@ let paren: 'a . ?first:space_formatter -> ?last:space_formatter ->
   = fun  ?(first=("": _ format6)) ?(last=("": _ format6)) b fu f x ->
     if b then (pp f "("; pp f first; fu f x; pp f last; pp f ")")
     else fu f x
+
 let constant_string f s = pp f "%S" s
+
 let tyvar f str = pp f "'%s" str
-let constant f = function
+
+(* In some places parens shouldn't be printed for readability:
+ * e.g. Some((-1)) should be printed as Some(-1)
+ * In `1 + (-1)` -1 should be wrapped in parens for readability
+ *)
+let constant ?(parens=true) f = function
   | Pconst_char i -> pp f "%C"  i
   | Pconst_string (i, None) -> pp f "%S" i
   | Pconst_string (i, Some delim) -> pp f "{%s|%s|%s}" delim i delim
-  | Pconst_integer (i, None) -> paren (i.[0]='-') (fun f -> pp f "%s") f i
+  | Pconst_integer (i, None) ->
+      let parens = if parens then (i.[0]='-') else false in
+      paren parens (fun f -> pp f "%s") f i
   | Pconst_integer (i, Some m) ->
-    paren (i.[0]='-') (fun f (i, m) -> pp f "%s%c" i m) f (i,m)
-  | Pconst_float (i, None) -> paren (i.[0]='-') (fun f -> pp f "%s") f i
-  | Pconst_float (i, Some m) -> paren (i.[0]='-') (fun f (i,m) ->
+      let parens = if parens then (i.[0]='-') else false in
+      paren parens (fun f (i, m) -> pp f "%s%c" i m) f (i,m)
+  | Pconst_float (i, None) ->
+      let parens = if parens then (i.[0]='-') else false in
+      paren parens (fun f -> pp f "%s") f i
+  | Pconst_float (i, Some m) ->
+      let parens = if parens then (i.[0]='-') else false in
```

Comment:
This can be simplified to `parens && i.[0]='-'`

Reply:
done